migration/src/main/java/kin/sdk/migration/sdk_related/KinAccountSdkImpl.java

```diff
@@ -44,42 +44,48 @@ public String getPublicAddress() {
+                                                   @NonNull BigDecimal amount, @NonNull IWhitelistService whitelistService) {
+        return sendTransaction(publicAddress, amount, whitelistService, null);
     }
 
     @NonNull
     @Override
-    public Request<ITransactionId> sendTransaction(final @NonNull String publicAddress,
-            final @NonNull BigDecimal amount, final @Nullable String memo) {
+    public Request<ITransactionId> sendTransaction(final @NonNull String publicAddress, final @NonNull BigDecimal amount,
+                                                   final @NonNull IWhitelistService whitelistService, final @Nullable String memo) {
         return new Request<>(new Callable<ITransactionId>() {
             @Override
             public ITransactionId call() throws Exception {
-                return sendTransactionSync(publicAddress, amount, memo);
+                return sendTransactionSync(publicAddress, amount, whitelistService, memo);
             }
         });
     }
 
     @NonNull
     @Override
-    public ITransactionId sendTransactionSync(@NonNull String publicAddress, @NonNull BigDecimal amount) throws OperationFailedException {
-        return sendTransactionSync(publicAddress, amount, null);
+    public ITransactionId sendTransactionSync(@NonNull String publicAddress, @NonNull BigDecimal amount,
+                                              @NonNull IWhitelistService whitelistService) throws OperationFailedException {
+        return sendTransactionSync(publicAddress, amount, whitelistService, null);
     }
 
     @NonNull
     @Override
-    public ITransactionId sendTransactionSync(@NonNull String publicAddress, @NonNull BigDecimal amount, @Nullable String memo) throws OperationFailedException {
+    public ITransactionId sendTransactionSync(@NonNull String publicAddress, @NonNull BigDecimal amount,
+                                              @NonNull IWhitelistService whitelistService, @Nullable String memo) throws OperationFailedException {
         try {
             Transaction transaction = kinAccount.buildTransactionSync(publicAddress, amount, 0, memo);
             if (whitelistService != null) {
-                String whitelistTransaction = whitelistService.whitelistTransaction(new KinSdkTransaction(transaction).getWhitelistableTransaction());
-                TransactionId transactionId = kinAccount.sendWhitelistTransactionSync(whitelistTransaction);
+                TransactionId transactionId = null;
+                WhitelistResult whitelistTransactionResult = whitelistService.onWhitelistableTransactionReady(new KinSdkTransaction(transaction).getWhitelistableTransaction());
+                if (whitelistTransactionResult.shouldSendTransaction()) {
+                    transactionId = kinAccount.sendWhitelistTransactionSync(whitelistTransactionResult.getWhitelistedTransaction());
+                }
                 return new KinSdkTransactionId(transactionId);
```

Comment:
probably a potential bug, in case we're not sending the transaction tx id will be null and can raise issue in the receiver side.

Reply:
We need to return something so if transaction id is null the developers should handle it.
The developer itself need to expect this because he was the one who decides not to send the transaction